src/assembler.jl

```diff
@@ -199,6 +199,20 @@ end
 end
 
 
+## assemble! with local condensation ##
+
+# TODO: assemble_zero! instead of last Bool arg?
+# TODO: Call this apply_assemble! instead?
```

Comment:
Thoughts? I don't see any real reason for making this a method of `assemble!`, in particular it isn't so obvious that something quite different is happening under the hood when passing the constraint handler as the second argument. Might as well make it clear by using a different function name altogether.

Reply:
`assemble_eliminated!` or `assemble_constrained!` maybe?

Reply:
It does `apply!` + `assemble!` which is what these operations are called in isolation in Ferrite so :shrug:

Reply:
👍 for not using `assemble!` for the combined function.
But wouldn't it be clearer to first call `apply_local*!` (mirroring the global versions) and then `assemble!`, instead of having one function that does both?

Reply:
The PR includes `apply_local!` too, but that only works if all constraints are local (e.g. Dirichlet), otherwise you need to write to the global matrix at positions outside the local matrix.

Reply:
Do you mean that exporting this call might be a bit misleading to export as it also modifies `Kglobal` and `fglobal`:
`_apply_local!(local_matrix, local_vector, global_dofs, ch, apply_zero, Kglobal, fglobal)`?
(I suppose it would be simplified to `apply[_zero]_local!(local_matrix, local_vector, global_dofs, ch, assembler)`, but it still is not purely local?)

Reply:
I feel it is a bit weird to have parts of the assembly happen in `apply_local!` and parts in the following `assemble!` call.

Reply:
Right, sorry! I didn't think of what happens in `apply_local!` as a type of assembly, but it is and then it makes perfect sense to combine it into one! 
I would then vote for removing the bool + using the `apply_assemble!`/`apply_zero_assemble` versions.